### ISSUE TITLE
some locales like ko-kr, ja-jp don't uses fractions in currency but angular does

### ISSUE THREAD

**Issue by clichedmoog** [angular/angular.js#13756]
How about adjust minFrac value in NUMBER_FORMATS for currency like MS excel does.
So it would be $filter(currency)(1000) = ¥1,000 and  $filter(currency)(1000.1) = ¥1,000.1 in ja-jp locale 

for reference: https://en.wikipedia.org/wiki/ISO_4217#Active_codes

**Comment by Narretz** [comment]
WE get the locale data from the Google closure i18n project - any change requests should be directed there. If it's fixed in closure, then we need to re-import the locale data.

**Comment by clichedmoog** [comment]
Thanks, I'll report them.

**Comment by clichedmoog** [comment]
I've reopen this issue because using the CLDR for JPY, KRW the fraction should be ignored.

I reported to closure repo
https://github.com/google/closure-library/issues/636

And they are following the Unicode CLDR, I've reported also.
https://unicode.org/cldr/trac/ticket/9230
As a result in KRW, JPY fractions should be ignored in currency notation.
https://unicode.org/reports/tr35/tr35-numbers.html#Number_Patterns

In my opinion, minFrac should follow the first value of goog.i18n.currency.CurrencyInfo in the closure lib's currency.js file.
https://github.com/google/closure-library/blob/master/closure/goog/i18n/currency.js

```
'JPY': [0, '¥', 'JP¥'],
'KRW': [0, '\u20A9', 'KR₩'],
```

**Comment by clichedmoog** [comment]
I've reopen this issue because using the CLDR for JPY, KRW the fraction should be ignored.

I reported to closure repo
google/closure-library#636

And they are following the Unicode CLDR, I've reported also.
https://unicode.org/cldr/trac/ticket/9230
As a result in KRW, JPY fractions should be ignored in currency notation.
https://unicode.org/reports/tr35/tr35-numbers.html#Number_Patterns

In my opinion, minFrac should follow the first value of goog.i18n.currency.CurrencyInfo in the closure lib's currency.js file.
https://github.com/google/closure-library/blob/master/closure/goog/i18n/currency.js

```
'JPY': [0, '¥', 'JP¥'],
'KRW': [0, '\u20A9', 'KR₩'],
```

**Comment by gkalpak** [comment]
Has the Google Closure i18n project fixed this? It is hard to tell by looking at the issues.

**Comment by clichedmoog** [comment]
@gkalpak no they did not fix, but as I think by their description Google's closure-i18n implemented well.
I'll investigate this more which library is causing the problem. so far I've searched CLDR is not the problem.
But I have only small information on this so which part of Google's closure-i18n project affects on minFrac value for currency filter?

**Comment by clichedmoog** [comment]
I've checked Google Closure i18n's implemention on KRW, JPY
Tested version: google-closure-library@20180805.0.0
Tested code:
```
require("google-closure-library");
goog.require('goog.i18n.NumberFormat');
var formatter_1 = new goog.i18n.NumberFormat(goog.i18n.NumberFormat.Format.CURRENCY, 'USD');
console.log(formatter_1.format(15650.579));
var formatter_2 = new goog.i18n.NumberFormat(goog.i18n.NumberFormat.Format.CURRENCY, 'KRW');
console.log(formatter_2.format(15650.579));
var formatter_3 = new goog.i18n.NumberFormat(goog.i18n.NumberFormat.Format.CURRENCY, 'JPY');
console.log(formatter_3.format(15650.579));
```

Results:
```
$15,650.58
₩15,651
¥15,651
```

**Comment by clichedmoog** [comment]
Oh, angular.js fixed it recently (not fixed on 1.6.4 and fixed from 1.6.5)
https://code.angularjs.org/1.6.5/i18n/angular-locale_ko-kr.js
https://code.angularjs.org/1.6.5/i18n/angular-locale_ja-jp.js

For KRW, JPY currency minFrac is 0 now. 
It is strange  currency setting not changed by CLDR to v30.0.1 update.
But I am so happy it's fixed now. 😀 

And I'am using 1.2.x for compability issue.
Would you reflect the changes to all versions please?

**Comment by gkalpak** [comment]
Aha! Glad we've fixed it :smiley:

We don't backport any changes other than security fixes to 1.2.x. See our version support policy [here][1].
In fact, AngularJS entered a [3 year Long Term Support][2] period on July 1, 2018

[1]: https://docs.angularjs.org/misc/version-support-status
[2]: https://blog.angular.io/stable-angularjs-and-long-term-support-7e077635ee9c
